src/components/dataset/schemaCreation/DatasetSchemaInformationView.tsx

```diff
@@ -0,0 +1,342 @@
+    marginLeft: 14,
+  },
+});
+
+interface IProps extends WithStyles<typeof styles> {
+  userEmail: string;
+}
+
+const DatasetSchemaInformationView = withStyles(styles)(({ classes }: IProps) => {
+  const [regionOptions, setRegionOptions] = useState(CLOUD_PLATFORMS.gcp.regions);
+  const {
+    register,
+    control,
+    setValue,
+    formState: { errors },
+  } = useFormContext();
+
+  const editorOptions = useMemo(
+    () =>
+      ({
+        previewRender(markdownText) {
+          return ReactDOMServer.renderToString(
+            <WithoutStylesMarkdownContent markdownText={markdownText} />,
+          );
+        },
+        status: false,
+      } as SimpleMDE.Options),
+    [],
+  );
+
+  return (
+    <Grid container rowSpacing={2} columnSpacing={5} className={classes.contentContainer}>
+      <Grid item xs={12}>
+        <Typography variant="h3">Dataset information</Typography>
+        We need some basic information about the dataset
+      </Grid>
+      <Grid item xs={12} md={6} lg={6} className={classes.formFieldContainer}>
+        <label
+          htmlFor="dataset-name"
+          className={clsx(classes.formLabel, { [classes.formLabelError]: errors.name })}
+        >
+          Name your dataset*
+        </label>
+        <TextField
+          id="dataset-name"
+          className={classes.formInput}
+          placeholder="Dataset name"
+          variant="outlined"
+          error={!!errors.name}
+          helperText={errors.name ? errors.name.message : ''}
+          {...register('name', {
+            required: 'Name is required',
+            minLength: { value: 1, message: 'Name must be 1+ characters long' },
+            maxLength: { value: 511, message: 'Name must be less than 511 characters long' },
+            pattern: {
+              value: /^[a-zA-Z0-9][_a-zA-Z0-9]*$/,
+              message: 'Name should fit pattern: ^[a-zA-Z0-9][_a-zA-Z0-9]*$',
+            },
+          })}
+        />
+      </Grid>
+
+      <Grid item xs={12} className={classes.formFieldContainer}>
+        <label
+          htmlFor="dataset-description"
+          className={clsx(classes.formLabel, { [classes.formLabelError]: errors.description })}
+        >
+          Dataset description*
+        </label>
+        <Controller
+          name="description"
+          control={control}
+          rules={{
+            required: 'Description is required',
+            maxLength: { value: 10000, message: 'Description must be under 10,000 characters' },
+          }}
+          render={({ field }) => (
+            <>
+              <SimpleMdeReact
+                options={editorOptions}
+                {...field}
+                placeholder="Dataset description"
+              />
+              {errors.description && (
+                <span className={classes.formInputError}>{errors.description.message}</span>
+              )}
+            </>
+          )}
+        />
+      </Grid>
+
+      <Grid item xs={6}>
```

Comment:
It feels like a lot of the Grid item code is repeated. I wonder if that could be an abstracted component? Especially because this is a very meaty component right now, and I think it could easily end up with high complexity as things need to change.

Reply:
I am going to push back on that a little? Grid is already a component, I don't really see too much value in abstracting it out further, we would be trading lines of code for lines of config imo. I prefer not to make components unless they are really reuseable as well, and I don't think a Grid surrounding a single component will be

Reply:
My biggest concern is the repeated code of the children. For example:
```
<label
          htmlFor="{FIELD}"
          className={clsx(classes.formLabel, { [classes.formLabelError]: {ERROR CLASS} })}
        >
          {LABEL DISPLAY}
        </label>
```
Is almost verbatim repeated across every grid component. However, especially since its not my codebase, if you think that the value gain is negligible, I am totally okay being overruled here. I just always find it a bit of a yellow flag when I see a lot of repeated code like this. (Another option would be to pull form fields out into a list of objects, and then map over that list to create the grid elements, but I think I also defer to the folks working in the codebase for style preferences)

Reply:
In this case, I also want to defer to the tech leads in the project too 😂 I'm also interested in hearing other opinions too! Maybe I'm off base here! :)